packages/git-diff/lib/git-diff-view.js

```diff
@@ -72,122 +151,106 @@ module.exports = class GitDiffView {
     this.moveToLineNumber(nextDiffLineNumber);
   }
 
-  updateIconDecoration() {
-    const gutter = this.editor.getElement().querySelector('.gutter');
-    if (gutter) {
-      if (
-        atom.config.get('editor.showLineNumbers') &&
-        atom.config.get('git-diff.showIconsInEditorGutter')
-      ) {
-        gutter.classList.add('git-diff-icon');
-      } else {
-        gutter.classList.remove('git-diff-icon');
-      }
-    }
-  }
-
   moveToPreviousDiff() {
     const cursorLineNumber = this.editor.getCursorBufferPosition().row + 1;
-    let previousDiffLineNumber = -1;
-    let lastDiffLineNumber = -1;
-    if (this.diffs) {
-      for (const { newStart } of this.diffs) {
-        if (newStart < cursorLineNumber) {
-          previousDiffLineNumber = Math.max(
-            newStart - 1,
-            previousDiffLineNumber
-          );
-        }
-        lastDiffLineNumber = Math.max(newStart - 1, lastDiffLineNumber);
+    let previousDiffLineNumber = null;
+    let lastDiffLineNumber = null;
+    for (const { newStart } of this.diffs) {
+      if (newStart < cursorLineNumber) {
+        previousDiffLineNumber = Math.max(newStart - 1, previousDiffLineNumber);
       }
+      lastDiffLineNumber = Math.max(newStart - 1, lastDiffLineNumber);
     }
 
     // Wrap around to the last diff in the file
     if (
       atom.config.get('git-diff.wrapAroundOnMoveToDiff') &&
-      previousDiffLineNumber === -1
+      previousDiffLineNumber === null
     ) {
       previousDiffLineNumber = lastDiffLineNumber;
     }
 
     this.moveToLineNumber(previousDiffLineNumber);
   }
 
-  moveToLineNumber(lineNumber) {
-    if (lineNumber != null && lineNumber >= 0) {
-      this.editor.setCursorBufferPosition([lineNumber, 0]);
-      this.editor.moveToFirstCharacterOfLine();
+  updateIconDecoration() {
+    const editorElement = atom.views.getView(this.editor);
```

Comment:
I see `atom.views.getView` many times. Please cache it in `this.editorElement`

Reply:
Okay